MediaBrowser.Providers/Plugins/MusicBrainz/MusicBrainzAlbumProvider.cs

```diff
@@ -256,6 +256,54 @@ public async Task<MetadataResult<MusicAlbum>> GetMetadata(AlbumInfo info, Cancel
             }
         }
 
+        if (Plugin.Instance != null && Plugin.Instance.Configuration.GetMissingTrackInfo)
+        {
+            var children = result.Item.Children.ToArray();
+            if (children.Any(s => s is Audio && s.IndexNumber == null) && releaseId != null)
+            {
+                var rspobj = _musicBrainzQuery.LookupRelease(Guid.Parse(releaseId), Include.Recordings);
+
+                IList<ITrack> tracks = new List<ITrack>();
+
+                if (rspobj.Media != null)
+                {
+                    foreach (var t in rspobj.Media.Where(x => x.Tracks != null).SelectMany(x => x.Tracks!))
+                    {
+                        tracks.Add(t);
+                    }
+                }
+
+                foreach (var c in children.Where(x =>
+                    x is Audio &&
+                    children.Any(p => Guid.Equals(p.Id, x.ParentId)) &&
+                    string.Equals(x.MediaType, MediaType.Audio, StringComparison.OrdinalIgnoreCase) &&
+                    ((x.RunTimeTicks == null) || (x.IndexNumber == null))))
+                {
+                    long? ticks = c.RunTimeTicks;
+                    int? index = c.IndexNumber;
+                    var trk = tracks.FirstOrDefault(t => string.Equals(t.Title, c.Name, StringComparison.OrdinalIgnoreCase));
+                    if (c.RunTimeTicks == null)
+                    {
+                        c.RunTimeTicks = trk?.Length?.Ticks ?? 0L;
+                    }
+
+                    if (c.IndexNumber == null)
+                    {
+                        c.IndexNumber = trk?.Position;
+                    }
+
+                    if ((ticks != c.RunTimeTicks) || (index != c.IndexNumber))
+                    {
+                        if (children.Where(p => Guid.Equals(p.Id, c.ParentId)).FirstOrDefault() is MusicAlbum parent)
```

Comment:
Can this ever happen. A MusicAlbum is not Audio, so there should be no musicalbums in children

Reply:
Yes, because this loop is filtered by instances where `RunTimeTicks` or `IndexNumber` is `null` meaning that this metadata is missing.